crates/sui-core/src/authority/authority_per_epoch_store.rs

```diff
@@ -1384,7 +1412,16 @@ impl AuthorityPerEpochStore {
             Err(e) => return Err(e),
         };
         let mut batch = tables.pending_execution.batch();
+        // pending_execution stores transactions received from consensus which may not have
+        // been executed yet. At this point, they have been committed to the db durably and
+        // can be removed.
+        // After end-to-end quarantining, we will not need pending_execution since the consensus
+        // log itself will be used for recovery.
         batch.delete_batch(&tables.pending_execution, digests)?;
+
+        // Now that the transaction effects are committed, we will never re-execute, so we
+        // don't need to worry about equivocating.
+        batch.delete_batch(&tables.signed_effects_digests, digests)?;
```

Comment:
The epoch store will be dropped eventually anyway. Why bothering deleting this?

Reply:
If we serve reads from the db, having a smaller table will keep the reads faster. If we serve the reads above from cache, we will have to read the entire table in to memory on startup, in which case keeping it small will be a necessity.